### PR TITLE
SSCS-10057 implement scheduling & listing get hearing endpoint

### PR DESCRIPTION
**Pull request by bensonlievoco**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-10057

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/model/IndividualDetails.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class IndividualDetails {
+
+    private String title;
+    private String firstName;
+    private String lastName;
+    private String preferredHearingChannel;
+    private String interpreterLanguage;
+    private String reasonableAdjustments;
+    private  boolean vulnerableFlag;
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/IndividualDetails.java:18]: Weird formatting

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingApi.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package uk.gov.hmcts.reform.sscs.service;
+
+import feign.codec.Decoder;
+import feign.codec.StringDecoder;
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingApi.java:8]: How do we feel about wildcard imports @dattipoeHMCTS @TeaWithLucas ?

**Review by wmarshid — APPROVED** [review]
Looks good to me.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingApi.java (hunk 2 above)
Comment (dattipoeHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingApi.java:8]: If there are 5+ imports from the same package it's fine. Good spot

**Comment by dattipoeHMCTS** [comment]
@TeaWithLucas @wmarshid What do you guys think about setting up mock responses for tests a this stage?
@bensonlievoco Could you also add some validation to the beans? If you check in the Hearings chat we discussed using Javax bean validation

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingApi.java (hunk 2 above)
Comment (bensonlievoco) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingApi.java:8]: import org.springframework.*;  ?

**Comment by bensonlievoco** [comment]
> @TeaWithLucas @wmarshid What do you guys think about setting up mock responses for tests a this stage? @bensonlievoco Could you also add some validation to the beans? If you check in the Hearings chat we discussed using Javax bean validation



> @TeaWithLucas @wmarshid What do you guys think about setting up mock responses for tests a this stage? @bensonlievoco Could you also add some validation to the beans? If you check in the Hearings chat we discussed using Javax bean validation

Yes, sure, I am on it

File: src/main/java/uk/gov/hmcts/reform/sscs/model/Attendees.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.hmcts.reform.sscs.model;
```
[the remark this answers is not in the document]
Comment (wmarshid) [reply src/main/java/uk/gov/hmcts/reform/sscs/model/Attendees.java:1]: HMC = Hearings Management Component - it is what these clients will be connecting to afaik

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingApi.java (hunk 2 above)
Comment (TeaWithLucas) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingApi.java:10]: The checkstyle does not like wild imports...

**Comment by bensonlievoco** [comment]
this is the new one

https://github.com/hmcts/sscs-hearings-api/pull/39

**Pull request closed without merging**